digitalocean/datasource_digitalocean_firewall.go

```diff
@@ -0,0 +1,27 @@
+package digitalocean
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDigitalOceanFirewall() *schema.Resource {
+	fwSchema := firewallSchema()
+
+	for _, f := range fwSchema {
+		if !f.Required {
+			f.Computed = true
+		}
+	}
```

Comment:
```suggestion
	for _, f := range fwSchema {
		f.Computed = true
		f.Required = false
	}

	fwSchema["name"].ValidateFunc = nil

	fwSchema["firewall_id"] = &schema.Schema{
		Type:     schema.TypeString,
		Required: true,
	}
```

In order to make `name` not required, we can make some additional adjustments to the schema in the data source. We can mark everything as `Computed = true` and `Required = false` as their values will be assigned from the API. We also set the `ValidateFunc` for `name` to `nil` as the SDK complains that "ValidateFunc is for validating user input, there's nothing to validate on computed-only field."

Since `firewall_id` is only needed for the data source, we can add it here instead of the main schema and make it required.

Reply:
Ah. I see.